budcamp/Homework2/Homework2/Program.cs

```diff
@@ -0,0 +1,168 @@
+            markerBlue ??= new Marker("default");
+            markerBlue?.PrintMarker();
+        }
+
+        private static void Task2()
+        {
+            Book bookBasic = new Book();
+            bookBasic.Title = "BASIC";
+            bookBasic.Author = "S.TROELSTRA";
+
+            Book bookC = new Book();
+            bookC.Title = "C+";
+            bookC.Author = "G.RTRTG";
+
+            bookBasic.PrintBookInfo();
+            bookC.PrintBookInfo();
+        }
+
+        private static void Task3()
+        {
+            try
+            {
+                Console.WriteLine("Input the number of elements to store in the array :");
+                int arraySize = int.Parse(Console.ReadLine());
+                int[] numbers = new int[arraySize];
+
+                for (int i = 0; i < numbers.Length; i++)
+                {
+                    Console.WriteLine($"pls enter your {i + 1} number");
+                    numbers[i] = int.Parse(Console.ReadLine());
+
+                    Console.Clear();
+                }
+
+                Console.WriteLine("The values store into the array are :");
+
+                foreach (int value in numbers)
+                {
+                    Console.Write(value + " ");
+                }
+
+                Console.WriteLine();
+
+                PrintReverseValues(numbers);
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Something goes wrong with you input data. Are you sure you entered an int?");
+            }
+        }
+
+        public static void PrintReverseValues(int[] inputArr)
+        {
+            Array.Reverse(inputArr);
+
+            Console.WriteLine("The values store into the array in reverse are :");
+
+            foreach (int value in inputArr)
+            {
+                Console.Write(value + " ");
+            }
+        }
+
+        private static void Task4()
+        {
+            Console.WriteLine("Input the number of elements to store in the array :");
+
+            try
+            {
+                int arraySize = Convert.ToInt32(Console.ReadLine());
+                int[] numbers = new int[arraySize];
+
+                for (int i = 0; i < numbers.Length; i++)
+                {
+                    Console.WriteLine($"pls enter your {i + 1} number");
+                    numbers[i] = Convert.ToInt32(Console.ReadLine());
+
+                    Console.Clear();
+                }
+
+                FindDuplicateElements(numbers);
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Something goes wrong with you input data.Are you sure you entered an int ?");
+            }
+        }
+
+        public static void FindDuplicateElements(int[] inputArr)
+        {
+            int duplicateMaxCount = 0;
+
+            duplicateMaxCount = inputArr.GroupBy(s => s)
+                .Select(a => new { Value = a.Key, Count = a.Count() })
```

Comment:
Why do you need `Value` here? You're not using it in other lambdas. Can't you just do `.Select(a => a.Count())`?

Reply:
srry, i was using it at first, then i changed code and forgot, that i don't need it

Reply:
Actully you don't even need create that anonymous type, you could do it as Medyk suggested, but it's that critical, just FYI